File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hash_map = {}

        for s in strs :
            key = ''.join(sorted(s))
            if key not in hash_map:
                hash_map[key] = []
            hash_map[key].append(s)

        return list(hash_map.values())

File: 0049-group-anagrams/0049-group-anagrams.rs
use std::collections::HashMap;
impl Solution {
    pub fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
        let mut anagrams: HashMap<String, Vec<String>> = HashMap::new();
    
    for s in strs {
        let mut chars: Vec<char> = s.chars().collect();
        chars.sort_unstable();
        let key: String = chars.into_iter().collect();
        
        anagrams.entry(key).or_insert(Vec::new()).push(s);
    }
    
    anagrams.into_iter().map(|(_, v)| v).collect()
    }
}